src/field_special_scene.c

```diff
@@ -0,0 +1,365 @@
+#include "global.h"
+#include "task.h"
+#include "sprite.h"
+#include "field_map_obj.h"
+#include "songs.h"
+#include "sound.h"
+#include "palette.h"
+#include "script.h"
+#include "vars.h"
+#include "event_data.h"
+#include "main.h"
+
+#define SECONDS(value) ((signed) (60.0 * value + 0.5))
+
+// porthole states
+enum
+{
+    INIT_PORTHOLE,
+    IDLE_CHECK,
+    EXECUTE_MOVEMENT,
+    EXIT_PORTHOLE,
+};
+
+extern void SetCameraPanning(s16 x, s16 y);
+extern void SetCameraPanningCallback(void ( *callback)());
+extern void InstallCameraPanAheadCallback();
+extern void pal_fill_black(void);
+extern void MapGridSetMetatileIdAt(s32 x, s32 y, u16 metatileId);
+extern void DrawWholeMapView();
+
+extern s8 gTruckCamera_HorizontalTable[];
+
+extern u8 gUnknown_0858E8AB[];
+extern u8 gUnknown_0858E8AD[];
+
+void Task_Truck3(u8);
+
+s16 GetTruckCameraBobbingY(int a1)
+{
+    if (!(a1 % 120))
+        return -1;
+    else if ((a1 % 10) <= 4)
+        return 1;
+
+    return 0;
+}
+
+s16 GetTruckBoxMovement(int a1) // for the box movement?
+{
+    if (!((a1 + 120) % 180))
+        return -1;
+
+    return 0;
+}
+
+// smh STILL BROKEN IN EMERALD
+void Task_Truck1(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    s16 cameraYpan;
+    s16 box1 = 0;
+    s16 box2 = 0;
+    s16 box3 = 0;
+    u8 mapNum, mapGroup;
+    register s16 zero asm("r4");
+
+    box1 = GetTruckBoxMovement(data[0] + 30) * 4; // top box.
+    sub_808E82C(1, gSaveBlock1Ptr->location.mapNum, gSaveBlock1Ptr->location.mapGroup, 3, box1 + 3);
+    box2 = GetTruckBoxMovement(data[0]) * 2; // bottom left box.
+    sub_808E82C(2, gSaveBlock1Ptr->location.mapNum, gSaveBlock1Ptr->location.mapGroup, 0, box2 - 3);
+    box3 = GetTruckBoxMovement(data[0]) * 4; // bottom right box.
+    mapNum = gSaveBlock1Ptr->location.mapNum;
+    mapGroup = gSaveBlock1Ptr->location.mapGroup;
+    zero = 0;
+    sub_808E82C(3, mapNum, mapGroup, -3, box3);
+
+    if (++data[0] == SECONDS(500)) // this will never run
+        data[0] = zero; // reset the timer if it gets stuck.
```

Comment:
Do either of these work instead of `zero`?
```c
data[0] -= data[0];
```
```c
data[0] -= SECONDS(500);
```

Reply:
tried both. the main problem is that zero (r4) is allocated in the middle of the function call, which is strange.